### PR TITLE
validate parser

### PR DESCRIPTION
**Pull request by abhidg**
Add parser validation to schema.

parser.schema.json is a JSON schema for the adtl parser files in TOML
format.  This adds autocompletion and validation with the EvenBetterTOML
extension for VS Code, with the schema comment at the top of parser
files:

```
#:schema ../../schemas/dev/parser.schema.json
```

### CODE REVIEW

File: schemas/dev/parser.schema.json [hunk 1]
```diff
@@ -0,0 +1,776 @@
...
+        },
+        "has_immunosuppression_therapy_treatment": {
+          "description": "Immunosuppression therapies treatment",
+          "$ref": "#/definitions/mapping"
+        },
+        "has_comorbidity_other": {
+          "description": "Other comorbidities",
+          "$ref": "#/definitions/mapping"
+        },
+        "has_died": {
```
Comment (ekamau) [inline schemas/dev/parser.schema.json:405]: 'has_died' will usually map to 'dsterm = 4'. i.e., true if dsterm = 4, false if dsterm != 4. How do we code this in the parser?

**Review by ekamau — COMMENTED** [review]
We are keeping 'outcome' and 'outcome_date' in both subject and visits table? or in one?

File: schemas/dev/parser.schema.json (hunk 1 above)
Comment (abhidg) [reply schemas/dev/parser.schema.json:405]: can use `values = { 4 = true, 1 = false, 3 = false, 2 = false }`, where 1--4 are dsterm values

**Comment by abhidg** [comment]
> We are keeping 'outcome' and 'outcome_date' in both subject and visits table? or in one?

Thanks, removed it from subject, it should only be in visit

**Pull request merged**